File: app/src/main/java/com/tregula/graedukacyjna/composition/module/ActivityBindingModule.java
package com.tregula.graedukacyjna.composition.module;

import com.tregula.graedukacyjna.MainActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityBindingModule {

    @ContributesAndroidInjector
    public abstract MainActivity bindMainActivity();
}

File: app/src/main/java/com/tregula/graedukacyjna/composition/module/ActivityBindingModule.kt
package com.tregula.graedukacyjna.composition.module

import com.tregula.graedukacyjna.MainActivity

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityBindingModule {

    @ContributesAndroidInjector(modules = [MainActivityModule::class, FragmentBindingModule::class])
    abstract fun provideMainActivity(): MainActivity
}
